projects/rocthrust/thrust/unique_ptr.h

```diff
@@ -112,25 +188,49 @@ struct default_delete<T[], std::enable_if_t<!std::is_trivially_destructible_v<T>
   size_t m_size;
 };
 
-// For arrays of trivially destructible element types we intentionally do NOT
-// store the element count. Their destruction is a no-op, so only the raw
-// deallocation is required. Omitting the size keeps this deleter
-// specialization an empty (zero-size) type, allowing unique_ptr<T[]>
-// instantiations that use it to remain a zero-cost abstraction (the deleter
-// need not increase the overall object size).
+/*! \brief Default deleter specialization for arrays of trivially destructible types.
+ *
+ *  For arrays of trivially destructible element types, this specialization intentionally
+ *  does NOT store the element count. Their destruction is a no-op, so only the raw
+ *  deallocation is required. Omitting the size keeps this deleter specialization an
+ *  empty (zero-size) type, allowing \p unique_ptr<T[]> instantiations that use it to
+ *  remain a zero-cost abstraction (the deleter need not increase the overall object size).
```

Comment:
*Technically* until the dust settles around `trivial_abi`, `unique_ptr<T[]>` is not _always_ a zero-cost abstraction, even for trivially destructible types as it may prevent a pointer from being passed in registers during function invocation. Also wouldn't hurt to elaborate a bit more that we're talking about increasing the size of the `unique_ptr<T[]>` compared to a raw pointer.
```suggestion
 *  remain a low/zero-cost abstraction (the default deleter need not increase the overall size of unique_ptr<T[]> above the size of a simple pointer `T *`).
```